### ISSUE TITLE
(Some) “Stealthy” Syndicate Items have unintentionally worse disguises

### ISSUE THREAD

**Issue by nok-ko** [space-wizards/space-station-14#30927]
#28688 unintentionally undisguises the energy dagger (and probably some other stealthy weapons, haven’t checked all of them):

`EnergySword` has a contraband component.

`EnergyDagger` inherits from `EnergySword` by way of `EnergyDaggerLoud`.
```
- type: entity
  name: pen
  parent: EnergyDaggerLoud
  id: EnergyDagger
  suffix: E-Dagger
  description: 'A dark ink pen.'
```

Result: 
![image](https://github.com/user-attachments/assets/f59705b6-3884-4f15-a2ed-4afd9761f1d2)

**Comment by Cojoke-dot** [comment]
Same thing with Rigged gloves, its kinda silly to be able to tell they are contra while they are on a person

![Screenshot 2024-08-12 002023](https://github.com/user-attachments/assets/12dec197-9278-400c-9bd8-d2db780eb1d0)

**Comment by nok-ko** [comment]
My proposed solution would be to add a field to `ContrabandComponent` that tracks whether it’s “obviously contraband,” which is either always set or is altered once someone uses the item to do a crime (unsheathing edagger, for instance). Maybe make it not a boolean but a bunch of flags or something, letting Syndies always know which pen is the most dangerous one in the room.

*Also* it would be nice to make non-crew players like mouse mobs not have the ability to detect contraband, but that’s a separate issue.

**Comment by Plykiya** [comment]
> My proposed solution would be to add a field to `ContrabandComponent` that tracks whether it’s “obviously contraband,” which is always set on items like e-swords, and is initially unset but gets altered once someone uses the item to do a crime (unsheathing an e-dagger, for instance). Maybe make it not a boolean but a bunch of flags or something, letting Syndies always know which pen is the most dangerous one in the room.
> 
> _Also_ it would be nice to make non-crew players like mouse mobs not have the ability to detect contraband, but that’s a separate issue.

ContrabandComponent is already how it does it

**Comment by UbaserB** [comment]
Solved by #30930 for the most part, there might be more but it's better to post a new issue for specifics.

**Comment by nok-ko** [comment]
> > My proposed solution would be to add a field to `ContrabandComponent` that tracks whether it’s “obviously contraband,” which is always set on items like e-swords, and is initially unset but gets altered once someone uses the item to do a crime (unsheathing an e-dagger, for instance). Maybe make it not a boolean but a bunch of flags or something, letting Syndies always know which pen is the most dangerous one in the room.
> > _Also_ it would be nice to make non-crew players like mouse mobs not have the ability to detect contraband, but that’s a separate issue.
> 
> things can get the comp when toggled with ComponentToggler

If we’re still planning on that airport scanning machine suggestion

> > My proposed solution would be to add a field to `ContrabandComponent` that tracks whether it’s “obviously contraband,” which is always set on items like e-swords, and is initially unset but gets altered once someone uses the item to do a crime (unsheathing an e-dagger, for instance). Maybe make it not a boolean but a bunch of flags or something, letting Syndies always know which pen is the most dangerous one in the room.
> > _Also_ it would be nice to make non-crew players like mouse mobs not have the ability to detect contraband, but that’s a separate issue.
> 
> things can get the comp when toggled with ComponentToggler

Is that enough to support the [proposed “airport scanner” feature](https://github.com/space-wizards/docs/pull/214/files/0eccad7c1d255d9e37928f4ec4d37b2ba2f7611e#r1589028073), if we’re still doing that?
